torch/distributed/_tensor/api.py

```diff
@@ -422,14 +422,16 @@ def redistribute(
         if placements is None:
             raise RuntimeError("placements is needed for redistribute!")
 
-        for placement in placements:
+        placements = list(placements)
+        for i, placement in enumerate(placements):
             if placement.is_partial():
                 raise RuntimeError(
                     "Can not redistribute to _Partial, _Partial is for internal use only!"
                 )
             elif isinstance(placement, Shard) and placement.dim < 0:
                 # normalize shard dim to be positive
-                placement.dim += self.ndim
+                placements[i] = Shard(placement.dim + self.ndim)
+        placements = tuple(placements)
```

Comment:
nit: we can remove the tuple inside the `Redistributed` Autograd function then? https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/redistribute.py#L190

We can also change the autograd function type annotation to align with tuple instead.